### DISCUSSION TITLE
x11 and wayfire and labwc requiring different customization methods?

### DISCUSSION THREAD

**Discussion by tinker2much** [gitbls/sdm#210] [General]
I've been doing some research on what config files are affected/required when you make various GUI tweaks - adding the cpu-temp widget, or adding a launcher, or activating a menu entry. X11 and wayfire sometimes use different config files to register such tweaks. And now there's lab-wc. 

At the moment, I collect ALL such configs, at least those required by X11 or wayfire, into a skeletal version of /home/pi/; that skeleton being dotted with files in various subdirectories like .config or .local., then I deploy the whole thing using the copydir plugin, i.e. rsync. So far, nothing seems to conflict. I was going to continue my research, seeing whether labwc required yet more configs, by booting an image burned this way, and bringing it up in labwc - if any of my desired tweaks don't appear, I'll make them manually, and see which _other_ file(s) have been touched. Then add THOSE to my skeleton too. 

Do you have any thoughts on this method, or any thoughts on how you would deal with this in sdm - i.e., accomplish the user's desired tweak whichever (compositor?) is in use?

**Comment by gitbls** [comment]
I think that's fgreat approach. I haven't looked at labwc at all (other than reading the feedback in the forum), but unless they use the same filenames/locations, you should be able to load up the configs for both wayfire and labwc without problems.

I use a plugin that used to be a Custom Phase Script. It does all that for me, and I'm loath to do anything about it because a) it works, and b) too many other higher payback (and more fun) things to do with my time.
